### PR TITLE
Added and reviewed separate() examples

### PR DESCRIPTION
**Pull request by jminnier**
Added a few examples using:

- convert=TRUE
- regex for column separations
- quasiquotation

#tidy-dev-day addresses issue #526

### CODE REVIEW

**Comment by hadley** [comment]
It looks like you have a few other Rd differences. Can you please check you have the latest roxygen2?

File: R/separate.R [hunk 1]
```diff
@@ -48,14 +48,42 @@
...
 #' # Another option:
 #' df %>% separate(x, c("a", "b"), extra = "merge", fill = "left")
+#' # Or you can keep all three
+#' df %>% separate(x, c("a", "b", "c"))
 #'
 #' # If only want to split specified number of times use extra = "merge"
 #' df <- data.frame(x = c("x: 123", "y: error: 7"))
 #' df %>% separate(x, c("key", "value"), ": ", extra = "merge")
+#'
+#' # Use regular expressions to separate on multiple characters:
```
Comment (hadley) [inline R/separate.R]: This is great!

File: R/separate.R [hunk 2]
```diff
@@ -48,14 +48,42 @@
...
+#' df <- data.frame(x = c(NA, "a?b", "a.d", "b:c"))
+#' df %>% separate(x, c("A","B"), sep = "([\\.\\?\\:])")
+#'
+#' # convert = TRUE detects column classes
+#' df <- data.frame(x = c("a:1", "a:2", "c:4", "d", NA))
+#' df %>% separate(x, c("key","value"), ":") %>% str
+#' df %>% separate(x, c("key","value"), ":", convert = TRUE) %>% str
+#'
+#' # Argument col can take quasiquotation
+#' quasi_commasep <- function(data, col_name) {
```
Comment (hadley) [inline R/separate.R]: I wonder if there's a way to make this simpler somehow. I'm worried it's a bit much to take in in an example.

Comment (jminnier) [reply R/separate.R]: I was thinking the same thing... maybe a function is too much and I should just use the !! directly. I'll try again!

File: R/separate.R [hunk 3]
```diff
@@ -48,14 +48,29 @@
...
+#' df %>% separate(x, c("A","B"), sep = "([\\.\\?\\:])")
+#'
+#' # convert = TRUE detects column classes
+#' df <- data.frame(x = c("a:1", "a:2", "c:4", "d", NA))
+#' df %>% separate(x, c("key","value"), ":") %>% str
+#' df %>% separate(x, c("key","value"), ":", convert = TRUE) %>% str
+#'
+#' # Argument col can take quasiquotation to work with strings
+#' var <- "x"
+#' df %>% separate(!!var, c("key","value"),":")
```
Comment (hadley) [inline R/separate.R]:
```suggestion
#' df %>% separate(!!var, c("key","value"), ":")
```

**Comment by hadley** [comment]
Thank you!!

**Pull request merged**